modules/gdscript/language_server/gdscript_language_protocol.h

```diff
@@ -31,33 +31,56 @@
 #ifndef GDSCRIPT_PROTOCAL_SERVER_H
 #define GDSCRIPT_PROTOCAL_SERVER_H
 
+#include "core/io/stream_peer.h"
+#include "core/io/stream_peer_tcp.h"
+#include "core/io/tcp_server.h"
 #include "gdscript_text_document.h"
 #include "gdscript_workspace.h"
 #include "lsp.hpp"
 #include "modules/jsonrpc/jsonrpc.h"
-#include "modules/websocket/websocket_peer.h"
-#include "modules/websocket/websocket_server.h"
+
+#define LSP_MAX_BUFFER_SIZE 4194304
+#define LSP_MAX_CLIENTS 8
 
 class GDScriptLanguageProtocol : public JSONRPC {
 	GDCLASS(GDScriptLanguageProtocol, JSONRPC)
 
+private:
+	class LSPeer : public Reference {
+	public:
+		Ref<StreamPeerTCP> connection;
+
+		uint8_t req_buf[LSP_MAX_BUFFER_SIZE];
+		int req_pos;
+		bool has_header;
+		bool has_content;
+		int content_length;
+		Vector<CharString> res_queue;
+		int res_sent;
+
+		Error handle_data();
+		Error send_data();
+
+		LSPeer();
+	};
+
 	enum LSPErrorCode {
 		RequestCancelled = -32800,
 		ContentModified = -32801,
 	};
 
 	static GDScriptLanguageProtocol *singleton;
 
-	HashMap<int, Ref<WebSocketPeer> > clients;
-	WebSocketServer *server;
-	int lastest_client_id;
+	HashMap<int, Ref<LSPeer> > clients;
+	Ref<TCP_Server> server;
+	int latest_client_id;
+	int next_client_id;
 
 	Ref<GDScriptTextDocument> text_document;
 	Ref<GDScriptWorkspace> workspace;
 
-	void on_data_received(int p_id);
-	void on_client_connected(int p_id, const String &p_protocal);
-	void on_client_disconnected(int p_id, bool p_was_clean_close);
+	Error on_client_connected();
+	void on_client_disconnected(int c_ind);
```

Comment:
Keep the code style p_client for the parmater name?

Reply:
Indeed, all parameters should use `p_` prefix and an explicit name (so `p_client` would be much better than `c_ind` here indeed).